File: stubs/java/src/main/java/com/beastmode/rmddd/core/ModuleCapability.java
package com.beastmode.rmddd.core;

import java.util.Objects;

/**
 * Represents a capability provided by a module.
 */
public class ModuleCapability {
    
    private final String name;
    private final String description;
    private final boolean available;
    private final String version;
    
    public ModuleCapability(String name, String description, boolean available, String version) {
        this.name = Objects.requireNonNull(name, "Name cannot be null");
        this.description = Objects.requireNonNull(description, "Description cannot be null");
        this.available = available;
        this.version = Objects.requireNonNull(version, "Version cannot be null");
    }
    
    public String getName() {
        return name;
    }
    
    public String getDescription() {
        return description;
    }
    
    public boolean isAvailable() {
        return available;
    }
    
    public String getVersion() {
        return version;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ModuleCapability that = (ModuleCapability) o;
        return available == that.available &&
               Objects.equals(name, that.name) &&
               Objects.equals(description, that.description) &&
               Objects.equals(version, that.version);
    }
    
    @Override
    public int hashCode() {
        return Objects.hash(name, description, available, version);
    }
    
    @Override
    public String toString() {
        return "ModuleCapability{" +
               "name='" + name + '\'' +
               ", description='" + description + '\'' +
               ", available=" + available +
               ", version='" + version + '\'' +
               '}';
    }
}

File: stubs/csharp/src/BeastMode.RmDdd/Core/ModuleCapability.cs
using System;

namespace BeastMode.RmDdd.Core
{
    /// <summary>
    /// Represents a capability provided by a module.
    /// </summary>
    public class ModuleCapability
    {
        public string Name { get; }
        public string Description { get; }
        public bool Available { get; }
        public string Version { get; }
        
        public ModuleCapability(string name, string description, bool available, string version)
        {
            Name = name ?? throw new ArgumentNullException(nameof(name));
            Description = description ?? throw new ArgumentNullException(nameof(description));
            Available = available;
            Version = version ?? throw new ArgumentNullException(nameof(version));
        }
        
        public override bool Equals(object? obj)
        {
            if (obj is not ModuleCapability other) return false;
            
            return Name == other.Name &&
                   Description == other.Description &&
                   Available == other.Available &&
                   Version == other.Version;
        }
        
        public override int GetHashCode()
        {
            return HashCode.Combine(Name, Description, Available, Version);
        }
        
        public override string ToString()
        {
            return $"ModuleCapability {{ Name = '{Name}', Description = '{Description}', Available = {Available}, Version = '{Version}' }}";
        }
    }
}